---
layout: page
title: Publications
permalink: /pubs/
---

<html>

<style>
  .abstract {
    color: darkslategray;
  }
  .intro{
        color: darkslategray;
        margin-bottom: 30px;
    }
</style>

<body onload="init()">
<!--- NOTE: when adding new publications, do not forget to add the call to toggleAbstract to the init() function. --->

<p class="intro"> List of peer-reviewed publications and pre-prints in reverse chronological order.
You can toggle abstracts by clicking the corresponding button and hide all abstracts by reloading the page.
</p>



<p>
Vatanen, T., Jabbar, K.S., Ruohtula, T., Honkanen, J., Avila-Pacheco, J.,
Siljander, H., <u>Stražar, M.</u>, Oikarinen, S., Hyöty, H., Ilonen, J., Mitchell,
C.M.  ... & Xavier, R. J. (2022). 

<b>Mobile genetic elements from the maternal microbiome shape infant
gut microbial assembly and metabolism.</b>
Cell, 185(26), pp.4921-4936.

    <a href="https://www.cell.com/cell/fulltext/S0092-8674(22)01467-2"> Link </a> |
    <a onClick="toggleAbstract('abs_edia'); return false;" href="fallback.html">Toggle abstract</a>
</p>


<div id="abs_edia" class="abstract">
<p>
The perinatal period represents a critical window for cognitive and immune
system development, promoted by maternal and infant gut microbiomes and their
metabolites. Here, we tracked the co-development of microbiomes and metabolomes
from late pregnancy to 1 year of age using longitudinal multi-omics data from a
cohort of 70 mother-infant dyads. We discovered large-scale mother-to-infant
interspecies transfer of mobile genetic elements, frequently involving genes
associated with diet-related adaptations. Infant gut metabolomes were less
diverse than maternal but featured hundreds of unique metabolites and
microbe-metabolite associations not detected in mothers. Metabolomes and serum
cytokine signatures of infants who received regular—but not extensively
hydrolyzed—formula were distinct from those of exclusively breastfed infants.
Taken together, our integrative analysis expands the concept of vertical
transmission of the gut microbiome and provides original insights into the
development of maternal and infant microbiomes and metabolomes during late
pregnancy and early life.
</p>
</div>







<p>
Vatanen, T., Ang, Q.Y., Siegwald, L., Sarker, S.A., Le Roy, C.I., Duboux, S.,
Delannoy-Bruno, O., Ngom-Bru, C., Boulangé, C.L., <u>Stražar, M.</u>,
Avila-Pacheco, J., ... & Xavier, R. J. (2022). 
<b>A distinct clade of Bifidobacterium longum in the gut
of Bangladeshi children thrives during weaning. </b>
Cell, 185(23), pp.4280-4297.
    <a href="https://www.cell.com/cell/fulltext/S0092-8674(22)01326-5"> Link </a> |
    <a onClick="toggleAbstract('abs_dhaka'); return false;" href="fallback.html">Toggle abstract</a>
</p>



<div id="abs_dhaka" class="abstract">
<p>
The gut microbiome has an important role in infant health and development. We
characterized the fecal microbiome and metabolome of 222 young children in
Dhaka, Bangladesh during the first two years of life. A distinct
<i>Bifidobacterium longum</i> clade expanded with introduction of solid foods and
harbored enzymes for utilizing both breast milk and solid food substrates. The
clade was highly prevalent in Bangladesh, present globally (at lower
prevalence), and correlated with many other gut taxa and metabolites,
indicating an important role in gut ecology. We also found that the B. longum
clades and associated metabolites were implicated in childhood diarrhea and
early growth, including positive associations between growth measures and B.
longum subsp. infantis, indolelactate and N-acetylglutamate. Our data
demonstrate geographic, cultural, seasonal, and ecological heterogeneity that
should be accounted for when identifying microbiome factors implicated in and
potentially benefiting infant development.
</p>
</div>




<p>
    <u>Stražar, M.*,</u> Mourits, V. P.*, Koeken, V. A., de Bree, L. C. J., Moorlag, S. J., Joosten, L. A., ... & Xavier, R. J. (2021).
    <b>The influence of the gut microbiome on BCG-induced trained immunity</b>. Genome Biology, 22(1), 1-22.

    <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02482-0"> Link </a> |
    <a onClick="toggleAbstract('abs_bcg'); return false;" href="fallback.html">Toggle abstract</a>

</p>

<div id="abs_bcg" class="abstract">
    <p>The bacillus Calmette-Guérin (BCG) vaccine protects against tuberculosis and heterologous infections but elicits
    high inter-individual variation in specific and nonspecific, or trained, immune responses. While the gut microbiome
    is increasingly recognized as an important modulator of vaccine responses and immunity in general, its potential role
        in BCG-induced protection is largely unknown.</p>

<p>Stool and blood were collected from 321 healthy adults before BCG vaccination, followed by blood sampling after
    2 weeks and 3 months. Metagenomics based on de novo genome assembly reveals 43 immunomodulatory taxa. The nonspecific,
    trained immune response is detected by altered production of cytokines IL-6, IL-1β, and TNF-α upon ex vivo blood
    restimulation with Staphylococcus aureus and negatively correlates with abundance of <i>Roseburia</i>. The specific response,
    measured by IFN-γ production upon Mycobacterium tuberculosis stimulation, is associated positively with <i>Ruminococcus</i> and
    <i>Eggerthella lenta</i>. The identified immunomodulatory taxa also have the strongest effects on circulating metabolites, with
    Roseburia affecting phenylalanine metabolism. This is corroborated by abundances of relevant enzymes, suggesting
    alternate phenylalanine metabolism modules are activated in a <i>Roseburia</i> species-dependent manner.</p>

<p>Variability in cytokine production after BCG vaccination is associated with the abundance of microbial genomes,
    which in turn affect or produce metabolites in circulation. <i>Roseburia</i> is found to alter both trained immune
    responses and phenylalanine metabolism, revealing microbes and microbial products that may alter BCG-induced
    immunity. Together, our findings contribute to the understanding of specific and trained immune responses after
    BCG vaccination.</p>
</div>



<p>
    <u>Stražar, M</u>., Temba, G. S., Vlamakis, H., Kullaya, V. I., Lyamuya, F., Mmbaga, B. T., ... Xavier, R. J. (2021). Gut
    <b>microbiome-mediated metabolism effects on immunity in rural and urban African populations</b>. Nature communications, 12(1), 1-15. 
<b>Nature Communication Editors’ Highlights pages </b> (showcase of the 50 best papers recently published in an area)
|
<a href="https://www.nature.com/articles/s41467-021-25213-2"> Link </a> |
<a onClick="toggleAbstract('abs_tanzania'); return false;" href="fallback.html">Toggle abstract</a>
</p>

<div id="abs_tanzania" class="abstract">
    <p>The human gut microbiota is increasingly recognized as an important factor in modulating innate and adaptive
        immunity through release of ligands and metabolites that translocate into circulation. Urbanizing African
        populations harbor large intestinal diversity due to a range of lifestyles, providing the necessary variation to
        gauge immunomodulatory factors. Here, we uncover a gradient of intestinal microbial compositions from rural through urban
        Tanzanian, towards European samples, manifested both in relative abundance and genomic variation observed in stool metagenomics.
        The rural population shows increased <i>Bacteroidetes</i>, led by <i>Prevotella copri</i>, but also presence of fungi.
        Measured ex vivo cytokine responses were significantly associated with 34 immunomodulatory microbes, which have
        a larger impact on circulating metabolites than non-significant microbes. Pathway effects on cytokines, notably
        TNF-α and IFN-γ, differential metabolome analysis and enzyme copy number enrichment converge on histidine and
        arginine metabolism as potential immunomodulatory pathways mediated by <i>Bifidobacterium longum</i>
        and <i>Akkermansia muciniphila</i>.</p>
</div>


<p>Poličar, P. G., <u>Stražar, M</u>., and Zupan, B. (2021).
<b>Embedding to reference t-SNE space addresses batch effects in single-cell classification</b>. Machine Learning, 1-20.
    |
<a href="https://link.springer.com/article/10.1007/s10994-021-06043-1"> Link </a> |
<a onClick="toggleAbstract('abs_tsne_springer'); return false;" href="fallback.html">Toggle abstract</a>
</p>
<div id="abs_tsne_springer" class="abstract">
    <p>Dimensionality reduction techniques, such as t-SNE, can construct informative visualizations of high-dimensional
        data. When jointly visualising multiple data sets, a straightforward application of these methods often fails;
        instead of revealing underlying classes, the resulting visualizations expose dataset-specific clusters. To
        circumvent these batch effects, we propose an embedding procedure that uses a t-SNE visualization constructed
        on a reference data set as a scaffold for embedding new data points. Each data instance from a new, unseen,
        secondary data is embedded independently and does not change the reference embedding. This prevents any
        interactions between instances in the secondary data and implicitly mitigates batch effects. We
        demonstrate the utility of this approach by analyzing six recently published single-cell gene
        expression data sets with up to tens of thousands of cells and thousands of genes. The batch effects in
        our studies are particularly strong as the data comes from different institutions using different experimental
        protocols. The visualizations constructed by our proposed approach are clear of batch effects, and the cells
        from secondary data sets correctly co-cluster with cells of the same type from the primary data. We also show
        the predictive power of our simple, visual classification approach in t-SNE space matches the accuracy of
        specialized machine learning techniques that consider the entire compendium of features that profile single
        cells.</p>
</div>




<p> Poličar, P. G., <u>Stražar, M.</u>, Zupan, B. (2019). <b>openTSNE: a modular Python library for t-SNE dimensionality reduction and embedding</b>. BioRxiv, 731877.
    |
<a href="https://www.biorxiv.org/content/10.1101/731877v1.abstract"> Link </a> |
<a onClick="toggleAbstract('abs_open_tsne'); return false;" href="fallback.html">Toggle abstract</a>
</p>
<div id="abs_open_tsne" class="abstract">
    <p>Point-based visualisations of large, multi-dimensional data from molecular biology can reveal meaningful clusters.
        One of the most popular techniques to construct such visualisations is t-distributed stochastic neighbor embedding
        (t-SNE), for which a number of extensions have recently been proposed to address issues of scalability and the quality
        of the resulting visualisations. We introduce openTSNE, a modular Python library that implements the core t-SNE
        algorithm and its extensions. The library is orders of magnitude faster than existing popular implementations,
        including those from scikit-learn. Unique to openTSNE is also the mapping of new data to existing embeddings,
        which can surprisingly assist in solving batch effects.</p>

    <p>openTSNE is available at <a href="https://github.com/pavlin-policar/openTSNE">https://github.com/pavlin-policar/openTSNE</a>.</p>
</div>

<p>
    Godec, P., Pančur, M., Ilenič, N., Čopar, A., <u>Stražar, M.</u>, Erjavec, A., ... Zupan, B. (2019).
    <b>Democratized image analytics by visual programming through integration of deep models and small-scale machine learning.</b>
    Nature communications, 10(1), 1-7.  |
<a href="https://www.nature.com/articles/s41467-019-12397-x"> Link </a> |
<a onClick="toggleAbstract('abs_orange_img'); return false;" href="fallback.html">Toggle abstract</a>
</p>
<div id="abs_orange_img" class="abstract">
    <p>Analysis of biomedical images requires computational expertize that are uncommon among biomedical scientists.
        Deep learning approaches for image analysis provide an opportunity to develop user-friendly tools for
        exploratory data analysis. Here, we use the visual programming toolbox Orange
        (<a href="http://orange.biolab.si">http://orange.biolab.si</a>)
        to simplify image analysis by integrating deep-learning embedding, machine learning procedures, and data
        visualization. Orange supports the construction of data analysis workflows by assembling components for
        data preprocessing, visualization, and modeling. We equipped Orange with components that use pre-trained
        deep convolutional networks to profile images with vectors of features. These vectors are used in image
        clustering and classification in a framework that enables mining of image sets for both novel and experienced
        users. We demonstrate the utility of the tool in image analysis of progenitor cells in mouse bone healing,
        identification of developmental competence in mouse oocytes, subcellular protein localization in yeast, and
        developmental morphology of social amoebae.</p>
</div>


<p>
    <u>Stražar, M.</u>, Žagar, L., Kokošar, J., Tanko, V., Erjavec, A., Poličar, P. G., ... Zupan, B. (2019).
    <b>scOrange—a tool for hands-on training of concepts from single-cell data analytics</b>. Bioinformatics, 35(14), i4-i12. |
<a href="https://academic.oup.com/bioinformatics/article/35/14/i4/5529249?login=true"> Link </a> |
<a onClick="toggleAbstract('abs_scorange'); return false;" href="fallback.html">Toggle abstract</a>
</p>
<div id="abs_scorange" class="abstract">
    <p>Single-cell RNA sequencing allows us to simultaneously profile the transcriptomes of thousands of
        cells and to indulge in exploring cell diversity, development and discovery of new molecular mechanisms.
        Analysis of scRNA data involves a combination of non-trivial steps from statistics, data visualization,
        bioinformatics and machine learning. Training molecular biologists in single-cell data analysis and
        empowering them to review and analyze their data can be challenging, both because of the complexity of the
        methods and the steep learning curve.</p>

    <p>We propose a workshop-style training in single-cell data analytics that relies on an explorative data analysis
        toolbox and a hands-on teaching style. The training relies on scOrange, a newly developed extension of a data
        mining framework that features workflow design through visual programming and interactive visualizations.
        Workshops with scOrange can proceed much faster than similar training methods that rely on computer programming
        and analysis through scripting in R or Python, allowing the trainer to cover more ground in the same time-frame.
        We here review the design principles of the scOrange toolbox that support such workshops and propose a syllabus
        for the course. We also provide examples of data analysis workflows that instructors can use during the training.</p>

    <p>scOrange is an open-source software. The software, documentation and an emerging set of educational videos
        are available at <a href="http://singlecell.biolab.si">http://singlecell.biolab.si</a>.</p>
</div>

<p><u>Stražar, M.</u>, Curk, T. (2019). <b>Approximate multiple kernel learning with least-angle regression</b>. Neurocomputing, 340, 245-258. |
<a href="
https://www.sciencedirect.com/science/article/pii/S0925231219302449"> Link </a> | 
<a onClick="toggleAbstract('abs_mklaren'); return false;" href="fallback.html">Toggle abstract</a> |
<a href="/material/popups/pub_mklaren.html" target="_blank">Preview</a>
</p>

<div id="abs_mklaren" class="abstract">
<p>Kernel methods provide a principled way for general data representations. Multiple kernel learning and kernel approximation are
    often treated as separate tasks, with considerable savings in time and memory expected if the two are performed simultaneously.</p>

<p>Our proposed Mklaren algorithm selectively approximates multiple kernel matrices in regression. It uses Incomplete
Cholesky Decomposition and Least-angle regression (LAR) to select basis functions, achieving linear complexity both
in the number of data points and kernels. Since it approximates kernel matrices rather than functions, it allows to
combine an arbitrary set of kernels. Compared to single kernel-based approximations, it selectively approximates
    different kernels in different regions of the input spaces.</p>

<p>The LAR criterion provides a robust selection of inducing points in noisy settings, and an accurate modelling of
regression functions in continuous and discrete input spaces. Among general kernel matrix decompositions,
Mklaren achieves minimal approximation rank required for performance comparable to using the exact kernel matrix,
at a cost lower than 1% of required operations. Finally, we demonstrate the scalability and interpretability
in settings with millions of data points and thousands of kernels.</p>
</div>

<p><u>Stražar M.</u>, Ule J., Žitnik M, Zupan B., Curk T. (2016), <b>Orthogonal matrix
factorization enables integrative analysis of multiple RNA binding proteins.</b> Bioinformatics.
 |
<a href="
http://bioinformatics.oxfordjournals.org/content/early/2016/01/18/bioinformatics.btw003.abstract?keytype=ref&ijkey=7YhmkcWqEPHzqlb"> Link </a> | 
<a onClick="toggleAbstract('abs_ionmf'); return false;" href="fallback.html">Toggle abstract</a> |
<a href="/material/popups/pub_ionmf.html" target="_blank">Preview</a>
</p>

<div id="abs_ionmf" class="abstract">
<p>RNA binding proteins (RBPs) play important roles in post-transcriptional
control of gene expression, including splicing, transport, polyadenylation and
RNA stability. To model protein–RNA interactions by considering all available
sources of information, it is necessary to integrate the rapidly growing RBP
experimental data with the latest genome annotation, gene function, RNA
sequence and structure. Such integration is possible by matrix factorization,
where current approaches have an undesired tendency to identify only a small
number of the strongest patterns with overlapping features. Because protein–RNA
interactions are orchestrated by multiple factors, methods that identify
discriminative patterns of varying strengths are needed.</p>

<p>We have developed an integrative orthogonality-regularized
nonnegative matrix factorization (iONMF) to integrate multiple data sources and
discover non-overlapping, class-specific RNA binding patterns of varying
strengths. The orthogonality constraint halves the effective size of the factor
model and outperforms other NMF models in predicting RBP interaction sites on
RNA. We have integrated the largest data compendium to date, which includes 31
CLIP experiments on 19 RBPs involved in splicing (such as hnRNPs, U2AF2,
ELAVL1, TDP-43 and FUS) and processing of 3’UTR (Ago, IGF2BP). We show that the
integration of multiple data sources improves the predictive accuracy of
retrieval of RNA binding sites. In our study the key predictive factors of
protein–RNA interactions were the position of RNA structure and sequence
motifs, RBP co-binding and gene region type. We report on a number of
protein-specific patterns, many of which are consistent with experimentally
determined properties of RBPs.</p>
</div>


<p>
Kavšček, M., <u>Stražar, M.</u>, Curk, T., Natter, K.,  Petrovič, U. (2015). <b>Yeast as
a cell factory: current state and perspectives.</b> Microbial cell factories,
14(1), 1 |
<a href="http://microbialcellfactories.biomedcentral.com/articles/10.1186/s12934-015-0281-x">Link</a> |
<a onClick="toggleAbstract('abs_yeast'); return false;" href="fallback.html">Toggle abstract</a> |
<a href="/material/popups/pub_yeast.html" target="_blank">Preview</a>
</p>

<div id="abs_yeast" class="abstract">
<p>The yeast Saccharomyces cerevisiae is one of the oldest and most frequently
used microorganisms in biotechnology with successful applications in the
production of both bulk and fine chemicals. Yet, yeast researchers are faced
with the challenge to further its transition from the old workhorse to a modern
cell factory, fulfilling the requirements for next generation bioprocesses.
Many of the principles and tools that are applied for this development
originate from the field of synthetic biology and the engineered strains will
indeed be synthetic organisms. We provide an overview of the most important
aspects of this transition and highlight achievements in recent years as well
as trends in which yeast currently lags behind. These aspects include: the
enhancement of the substrate spectrum of yeast, with the focus on the efficient
utilization of renewable feedstocks, the enhancement of the product spectrum
through generation of independent circuits for the maintenance of redox
balances and biosynthesis of common carbon building blocks, the requirement for
accurate pathway control with improved genome editing and through orthogonal
promoters, and improvement of the tolerance of yeast for specific stress
conditions. The causative genetic elements for the required traits of the
future yeast cell factories will be assembled into genetic modules for fast
transfer between strains. These developments will benefit from progress in
bio-computational methods, which allow for the integration of different kinds
of data sets and algorithms, and from rapid advancement in genome editing,
which will enable multiplexed targeted integration of whole heterologous
pathways. The overall goal will be to provide a collection of modules and
circuits that work independently and can be combined at will, depending on the
individual conditions, and will result in an optimal synthetic host for a given
production process. </p>
</div>


<p>
<u>Stražar, M.</u>, Mraz, M., Zimic, N.,  Moškon, M. (2014). <b>An adaptive genetic
algorithm for parameter estimation of biological oscillator models to achieve
target quantitative system response.</b> Natural Computing, 13(1), 119-127 |
<a href="http://link.springer.com/article/10.1007/s11047-013-9383-8"> Link </a> |
<a onClick="toggleAbstract('abs_adaptive'); return false;" href="fallback.html">Toggle abstract</a> |
<a href="/material/popups/pub_adaptive.html" target="_blank">Preview</a>
</p>

<div id="abs_adaptive" class="abstract">
<p>Mathematical modeling has become an integral part of synthesizing gene
regulatory networks. One of the common problems is the determination of
parameters, which are a part of the model description. In the present work, we
propose a customized genetic algorithm as a method to determine the parameters
such that the underlying oscillatory system exhibits the target behavior. We
propose a problem specific, adaptive fitness function evaluation and a method
to quantify the effect of a single parameter on the system response. The
properties of the algorithm are highlighted and confirmed on two test cases of
synthetic biological oscillators.</p>
</div>


<p>
Lebar, T., Bezeljak, U., Golob, A., Jerala, M., Kadunc, L., Pirš, B., <u>Stražar M.</u>,
    ...  Jerala, R. (2014). <b>A bistable genetic switch based on designable
DNA-binding domains.</b> Nature communications, 5 |
<a href="http://www.nature.com/ncomms/2014/140929/ncomms6007/full/ncomms6007.html">Link</a> | 
<a onClick="toggleAbstract('abs_bistable'); return false;" href="fallback.html">Toggle abstract</a> |
<a href="/material/popups/pub_switch.html" target="_blank">Preview</a>
</p>

<div id="abs_bistable" class="abstract">
<p>Bistable switches are fundamental regulatory elements of complex systems,
ranging from electronics to living cells. Designed genetic toggle switches have
been constructed from pairs of natural transcriptional repressors wired to
inhibit one another. The complexity of the engineered regulatory circuits can
be increased using orthogonal transcriptional regulators based on designed
DNA-binding domains. However, a mutual repressor-based toggle switch comprising
DNA-binding domains of transcription-activator-like effectors (TALEs) did not
support bistability in mammalian cells. Here, the challenge of engineering a
bistable switch based on monomeric DNA-binding domains is solved via the
introduction of a positive feedback loop composed of activators based on the
same TALE domains as their opposing repressors and competition for the same DNA
operator site. This design introduces nonlinearity and results in epigenetic
bistability. This principle could be used to employ other monomeric DNA-binding
domains such as CRISPR for applications ranging from reprogramming cells to
building digital biological memory.</p>
</div>

<p>
    Moškon, M., Zimic, N., <u>Stražar, M.</u>,  Mraz, M. (2013). <b>Comparison of selected
performances of biological and electronic information processing structures.</b>
Przegląd Elektrotechniczny, 89 | 
<a href="https://www.infona.pl/resource/bwmeta1.element.baztech-8535e86f-3659-4526-a3a2-9f039df4bda8">Link</a> |
<a onClick="toggleAbstract('abs_comparison'); return false;" href="fallback.html">Toggle abstract</a> |
<a href="/material/popups/pub_przeglad.html" target="_blank">Preview</a>
</p>

<div id="abs_comparison" class="abstract">
<p>
We present the information processing perspective on biological systems.
Several metrics, similar to the ones used in digital electronic circuits, are
introduced. These metrics allow us to compare biological information processing
structures with their electronic counterparts, to define the ones with the best
dynamical properties, analyse their compatibility and most importantly,
automatize their design. Regarding the metric values obtained and used on a
simple example, target applications of synthetic information processing
biological structures are discussed.</p>
</div>


<h2>Conference presentations</h2>

<p class="intro">Presentations at conferences, meetings and workshops.</p>


<p>
Godec, P., Zupan B., Tanko V., Stražar M. (2022),
<b>Efficient Matching of Single Cells and Cell Types</b>, 
XAI-Healthcare eXplainable AI in Healthcare, 
1-day workshop June 11-14, 2022 in conjunction with IEEE ICHI 2022, Rochester, Minnesota, USA
</p>

<p>Stražar, M., Temba, G. S., Vlamakis, H., Kullaya, V. I., Lyamuya, F., Mmbaga, B. T., ... Xavier, R. J. (2021).
    <b>Urbanization-driven changes in the gut microbiota reveal immunomodulatory metabolites and pathways</b>,
    <ul>
    <li>Impact of Environmental Exposures on the Microbiome and Human Health, NIEHS Workshop, February 23, 2021. Virtual.</li>
    <li>Harnessing the Microbiome for Disease Prevention and Therapy, Keystone Symposium, January 18-20, 2021. Keystone, Co. USA.</li>
    </ul>
    <a href="../posters/poster_tanzania.pdf">PDF</a>
    | <a href="../posters/poster_tanzania.png">PNG</a> </p>

<p> Stražar, M. and Žagar, L. and Kokošar, J. and Tanko, V. and Poličar, P. and Erjavec, A. and Starič, A. and
Menon, V. and Chen, R. and Shaulsky, G. and Lemire, A. and Parikh, A. and Zupan, B. (2018).

    <b>scOrange: Single-Cell Data Mining for Everyone.</b>
    The 16th International NETTAB workshop.  October 22-24. Genova, Italy.
    | <a href="../posters/poster_nettab.pdf">PDF</a>
    | <a href="../posters/poster_nettab.png">PNG</a> </p>

<p> Stražar M., Ule J., Curk T. (2017). <b>csDEX: condition-specific
differential exon expression.</b> 25th annual International Conference on
Intelligent Systems for Molecuar Biology. European conference on Computational
Biology. Prague, Czech Republic | <a
href="../posters/poster_csdex.pdf">PDF</a> | <a
href="../posters/poster_csdex.png">PNG</a> </p>

<p> Stražar M., Curk T. (2016) <b>Learning the kernel matrix by predictive
low-rank approximations.</b> Machine Learning Summer School. Cadiz, Spain |
<a href="../posters/poster_mklaren.pdf">PDF</a> | 
<a href="../posters/poster_mklaren.png">PNG</a> </p>

<p> Stražar M., Žitnik M., Zupan B., Ule J., Curk T. (2015). <b>Orthogonal
nonnegative factorization-based analysis of nineteen protein-RNA interaction
CLIP data sets</b>. 23rd annual International Conference on Intelligent Systems
for Molecuar Biology. European conference on Computational Biology. Dublin,
Ireland | <a href="../posters/poster_ionmf.pdf">PDF</a> | <a
href="../posters/poster_ionmf.png">PNG</a> </p>

<script>
function toggleAbstract(id) {
    var x = document.getElementById(id);
    if (x.style.display === "none") {
        x.style.display = "block";
    } else {
        x.style.display = "none";
    }
}

function init(){
    toggleAbstract("abs_dhaka");
    toggleAbstract("abs_edia");
    toggleAbstract("abs_bcg");
	toggleAbstract("abs_tanzania");
	toggleAbstract("abs_tsne_springer");
	toggleAbstract("abs_open_tsne");
	toggleAbstract("abs_orange_img");
	toggleAbstract("abs_scorange");
    toggleAbstract("abs_mklaren");
    toggleAbstract("abs_ionmf");
    toggleAbstract('abs_yeast');
    toggleAbstract('abs_adaptive');
    toggleAbstract('abs_bistable');
    toggleAbstract('abs_comparison');
}

</script>

</body>
</html>

